### PR TITLE
Add range tests demonstrating boundaries

### PR DESCRIPTION
**Pull request by mark-moir**
This PR establishes some tests to demonstrate the expressiveness of ranges for range proofs, and also test that they behave as expected at the boundaries of ranges.

### CODE REVIEW

File: tests/range.rs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+// These tests are expected to fail (expected_to_fail argument is true)
+range_test_with!(out_of_range_below, 0, Some(1), Some(isize::MAX), true);
+range_test_with!(out_of_range_above, 1001, Some(0), Some(1000), true);
+
+fn test_range_proof_works(
+    val: isize,
+    lower: Option<isize>,
+    upper: Option<isize>,
+    expexcted_to_fail: bool,
```
[the remark this answers is not in the document]
Comment (mark-moir) [reply tests/range.rs]: Good catch, thanks.  FIXED.

**Review by swcurran — APPROVED** [review]
